common/SmokeTests/SmokeTest/Program.cs

```diff
@@ -1,37 +1,227 @@
-            Console.WriteLine();
-            Console.WriteLine("SMOKE TEST FOR TRACK 2 SAMPLES");
-
-            await IotHubConnectionTests.RunTests();
-
-            Console.WriteLine("SAMPLES TESTS COMPLETE");
-            Console.WriteLine();
-            return 0;
+            ++typeCount;
+            ProcessType(type);
         }
     }
+    catch (Exception ex)
+    {
+        ReportError();
+        LogError($"{Environment.NewLine }[{ assembly.FullName }] could not be fully processed.  Error: { ex }{ Environment.NewLine }");
+    }
+
+    LogInformation($"Completed assembly: [{ assembly.FullName }].  { typeCount } types inspected.{ Environment.NewLine }");
+}
+
+// End the test run and report on the state.
+
+LogInformation($"{ Environment.NewLine }Smoke Test run complete.");
+
+if (returnCode == 0)
+{
+    LogInformation("Passed.", ConsoleColor.Green);
+}
+else
+{
+    LogInformation("Failed.", ConsoleColor.Red);
 }
+
+return returnCode;
+
+// Local function definitions
+
+void ReportError() => returnCode = -1;
+
+void LogError(string message, ConsoleColor? color = default)
+{
+    Console.ForegroundColor = color ?? ConsoleColor.Red;
+    Console.Error.WriteLine(message);
+    Console.ResetColor();
+}
+
+void LogInformation(string message, ConsoleColor? color = default)
+{
+    Console.ForegroundColor = color ?? Console.ForegroundColor;
+    Console.WriteLine(message);
+    Console.ResetColor();
+}
+
+void ProcessType(Type type)
+{
+    // Inspect the members of the type, walking each to force metadata to be loaded.  If a parameterless
+    // constructor is found during the scan, take note for later use if it is public or belongs to an Azure
+    // client type.
+
+    var isConstructable = false;
+
+    try
+    {
+        foreach (var member in type.GetMembers(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance))
+        {
+            if ((member is ConstructorInfo constructor) && (constructor.GetParameters().Length == 0))
+            {
+                isConstructable = ((constructor.IsPublic) || (IsAzureClientType(type)));
+            }
+        }
+
+        // If the type is non-generic, non-abstract, and has a parameterless constructor, it should be constructed.
+
+        if ((!type.ContainsGenericParameters) && (!type.IsAbstract) && (isConstructable))
+        {
+            try
+            {
+                _ = Activator.CreateInstance(type, true);
+
+                if (IsAzureClientType(type))
+                {
+                    Console.WriteLine($"\tConstructed: '{ type.FullName }'");
+                }
+            }
+            catch (TargetInvocationException ex) when (ex.InnerException is NotImplementedException)
+            {
+                // Expected; this occurs when the parameterless constructor is present but has been marked obsolete.  Since we do
+                // not report types without a parameterless constructor, ignore the type and do not log.
+            }
```

Comment:
```suggestion
            catch (TargetInvocationException ex) when (ex.InnerException is NotImplementedException)
            {
                // Expected; this occurs when the parameterless constructor is present but has been marked obsolete.  Since we do
                // not report types without a parameterless constructor, ignore the type and do not log.
            }
            catch (TargetInvocationException ex) when (ex.InnerException is ArgumentException)
            {
                // Expected; this occurs when the parameterless constructor is present but the type cannot be constructed due to failing
                // an internal state validation.  Since we do not report types without a parameterless constructor, ignore the type
                // and do not log.
            }
```